runtime/sema/crypto_algorithm_types.go

```diff
@@ -0,0 +1,190 @@
+/*
+ * Cadence - The resource-oriented smart contract programming language
+ *
+ * Copyright 2019-2021 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sema
+
+import (
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/errors"
+)
+
+//go:generate go run golang.org/x/tools/cmd/stringer -type=SignatureAlgorithm
+//go:generate go run golang.org/x/tools/cmd/stringer -type=HashAlgorithm
+
+var SignatureAlgorithms = []NativeEnumCase{
+	SignatureAlgorithmECDSA_P256,
+	SignatureAlgorithmECDSA_Secp256k1,
+	SignatureAlgorithmBLSBLS12381,
+}
+
+var HashAlgorithms = []NativeEnumCase{
+	HashAlgorithmSHA2_256,
+	HashAlgorithmSHA2_384,
+	HashAlgorithmSHA3_256,
+	HashAlgorithmSHA3_384,
+	HashAlgorithmKMAC128,
+}
+
+var SignatureAlgorithmType = newNativeEnumType(SignatureAlgorithmTypeName, &UInt8Type{})
+
+type SignatureAlgorithm int
+
+const (
+	// Supported signing algorithms
+	SignatureAlgorithmECDSA_P256 SignatureAlgorithm = iota
+	SignatureAlgorithmECDSA_Secp256k1
+	SignatureAlgorithmBLSBLS12381
+)
+
+// Name returns the string representation of this signing algorithm.
+func (algo SignatureAlgorithm) Name() string {
+	switch algo {
+	case SignatureAlgorithmECDSA_P256:
+		return "ECDSA_P256"
+	case SignatureAlgorithmECDSA_Secp256k1:
+		return "ECDSA_Secp256k1"
+	case SignatureAlgorithmBLSBLS12381:
+		return "BLSBLS12381"
+	}
+
+	panic(errors.NewUnreachableError())
+}
+
+func (algo SignatureAlgorithm) RawValue() int {
+	return int(algo)
+}
+
+func (algo SignatureAlgorithm) DocString() string {
+	switch algo {
+	case SignatureAlgorithmECDSA_P256:
```

Comment:
Just a general comment about the signature algorithm naming in the PR. We could use a more consistent rule `<algoName>_<curveName>` or `<algoName><curveName>`. 
In our case `ECDSA` and `BLS` are the algo names, while `P256`, `Secp256k1` and `BLS12381` are the curve names.

Reply:
So we could choose between: 
 - `ECDSA_P256` , `BLS_BLS123281` ..
 - `ECDSAP256` , `BLSBLS123281` ..

Reply:
Good catch! I like the suggestion for `<algoName>_<curveName>`, looks more readable. Also consistent with your new crypto proposal.

Updated in https://github.com/onflow/cadence/pull/633/commits/a6f332cb95a402df32b6cadc46490734149ff194